flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterPartitionReleaseTest.java

```diff
@@ -166,10 +168,7 @@ private void testPartitionReleaseOrPromotionOnJobTermination(
             jobMasterGateway.updateTaskExecutionState(
                     new TaskExecutionState(
                             taskDeploymentDescriptor.getExecutionAttemptId(), finalExecutionState));
-
-            assertThat(
-                    taskExecutorCallSelector.apply(testSetup).get(),
-                    equalTo(testSetup.getTaskExecutorResourceID()));
+            assertThat(callSelector.apply(testSetup).get(), notNullValue());
```

Comment:
This verification looks a bit weird. I guess what it is currently checking that the future is completed normally.
Maybe we can manually add some partitions to the tracker before terminating the job, and check whether they are released on job termination.

Reply:
That's much better.